src/rgw/motr/gc/gc.cc

```diff
@@ -14,16 +14,59 @@
  */
 
 #include "motr/gc/gc.h"
+#include <cstdlib>
+#include <ctime>
 
 void *MotrGC::GCWorker::entry() {
   std::unique_lock<std::mutex> lk(lock);
   ldpp_dout(dpp, 10) << __func__ << ": " << gc_thread_prefix
     << worker_id << " started." << dendl;
 
+  // Get random number to lock the GC index.
+  uint32_t my_index = ceph::util::generate_random_number() % (motr_gc->max_indices);
```

Comment:
`generate_random_number()` takes the range [start, end] as input I think. Without range, this will be similar to that of `rand()`